test-data/unit/check-namedtuple.test

```diff
@@ -691,6 +691,23 @@ my_eval(A([B(1), B(2)])) # OK
 [builtins fixtures/isinstancelist.pyi]
 [out]
 
+[case testNamedTupleImportCycle]
+import b
+[file a.py]
+class C:
+    pass
+
+from b import tp
+x: tp
+reveal_type(x.x)  # E: Revealed type is 'builtins.int'
```

Comment:
What if you have `tp('x')` in this file? Or `reveal_type(tp)`?

Reply:
Yes, this will not work. I was thinking about the `Invalid type` problem. All references in type context will be correctly re-resolved in the third pass. While there is no such thing as forward reference in runtime context.
I however still think this fix may have some value:
* We have seen many complains about `Invalid type`, but few complains about `Cannot determine type` (the error that appears if you try `tp('x')`)
* The reason for above fact is that many of the import cycle exist _only_ for annotations, `tp('x')` will fail at runtime  anyway
* This error quite often appears in stub files for large libraries, where we only care about type context

It is up to you of course.